tcms/search/advanced_search.py

```diff
@@ -78,43 +85,52 @@ def _sum_orm_queries(plans, cases, runs, target):
     cases = cases.evaluate()
     runs = runs.evaluate()
     if target == 'run':
-        if not plans and not cases:
-            if runs is None:
-                runs = TestRun.objects.none()
-        if runs is None:
-            runs = TestRun.objects.all()
-        if cases:
-            runs = runs.filter(case_run__case__in=cases).distinct()
-        if plans:
-            runs = runs.filter(plan__in=plans).distinct()
-        return runs
+        return _get_run_target(plans, cases, runs)
     if target == 'plan':
-        if not cases and not runs:
-            if plans is None:
-                plans = TestPlan.objects.none()
-        if plans is None:
-            plans = TestPlan.objects.all()
-        if cases:
-            plans = plans.filter(case__in=cases).distinct()
-        if runs:
-            plans = plans.filter(run__in=runs).distinct()
-        plans = plans.annotate(num_cases=Count('case', distinct=True),
-                               num_runs=Count('run', distinct=True))
-        return plans
+        return _get_plan_target(plans, cases, runs)
     if target == 'case':
-        if not plans and not runs:
-            if cases is None:
-                cases = TestCase.objects.none()
-        if cases is None:
-            cases = TestCase.objects.all()
-        if runs:
-            cases = cases.filter(case_run__run__in=runs).distinct()
-        if plans:
-            cases = cases.filter(plan__in=plans).distinct()
-        return cases
+        return _get_case_target(plans, cases, runs)
     raise ValueError('Invalid target')
 
 
+def _get_run_target(plans, cases, runs):
```

Comment:
Why do you move these lines into their own functions? What do we gain from that ?

Reply:
Else pylint throws a too-many-branches warning.